### PR TITLE
Hosting Command Palette: Add dynamic placeholders to nested commands

### PR DESCRIPTION
**Pull request by katinthehatsite**
Related:
* https://github.com/Automattic/wp-calypso/pull/84633
* Also specificallyhttps://github.com/Automattic/wp-calypso/pull/84633#discussion_r1408992884

## Proposed Changes

This PR adds a logic for implementing dynamic placeholders that can be set for each command separately. If there is no placeholder set, the placeholder will default to `Select a site`.

At the moment, I added the placeholder only to the `Open site dashboard` command to be `Select dashboard to open` just to implement the logic. As per the discussion with @sejas, we can adjust the wording per each command later. 

## Testing Instructions

* Navigate to Calypso Live Link
* Navigate to `/sites`
* Press `cmd + k` on macOS or `ctrl + k` on Windows to open the command palette
* Observe that the placeholder on the top says `Search for commands` and **IS NOT** changed
* Click on `Open site dashboard` command
* Observe that the placeholder on the top has been changed to `Select dashboard to open`

Screenshot 2023-11-29 at 11 35 13 AM

* Press `Esc` to go back to the top-level commands
* Confirm that placeholder is set back to `Search for commands`
* Select a command that has no custom placeholder set such as `Manage DNS`
* Confirm that the placeholder displayed is set to a default `Select a site`:

Screenshot 2023-11-29 at 11 34 43 AM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by katinthehatsite** [comment]
@sejas happy to add whatever text you would like to each command but we can also add it later.

File: client/sites-dashboard/components/wpcom-smp-commands.tsx [hunk 1]
```diff
@@ -32,13 +31,12 @@ interface useCommandsArrayWpcomOptions {
 export const useCommandsArrayWpcom = ( {
 	setSelectedCommandName,
 }: useCommandsArrayWpcomOptions ) => {
-	const translate = useTranslate();
 	const setStateCallback =
-		( actionName: string ) =>
+		( actionName: string, placeholder: string = 'Select a site' ) =>
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/wpcom-smp-commands.tsx]: Can you make sure this placeholder is translated?

Comment (katinthehatsite) [reply client/sites-dashboard/components/wpcom-smp-commands.tsx]: done!

**Review by sejas — APPROVED** [review]
Tested with `Open site dashboard` and I confirm we can add a different placeholder easily.

## Screencast

https://github.com/Automattic/wp-calypso/assets/779993/9656fad4-cd6d-47dd-9345-aa4adf28df64

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9782708

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @katinthehatsite for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9782708

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @katinthehatsite for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9782708).

**Pull request merged**